src/app/qgsversionmigration.h

```diff
@@ -0,0 +1,66 @@
+/***************************************************************************
+  qgsversionmigration.h - QgsVersionMigration
+
+ ---------------------
+ begin                : 30.7.2017
+ copyright            : (C) 2017 by nathan
+ email                : woodrow.nathan at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSVERSIONMIGRATION_H
+#define QGSVERSIONMIGRATION_H
+
+#include "qgis_app.h"
+#include "qgserror.h"
+
+class QgsSettings;
+
+/**
+ * Version migration class used to transfer settings, etc between major versions.
+ * \note Not everything can be translated and depends on the from and to versions.
+ */
+class APP_EXPORT QgsVersionMigration
+{
+  public:
+    QgsVersionMigration();
+    virtual ~QgsVersionMigration();
+
+    /**
+     * Check if two version has a migration options.
+     * @param fromVersion The version migrating from.
```

Comment:
@param/@return -> \param \return

Also better put a note about what this returns here